### ISSUE TITLE
Modeling CTC expansion giving weird phaseouts?

### ISSUE THREAD

**Issue by JackLandry** [PSLmodels/Tax-Calculator#2614]
I am trying to model the effects of the ARPA CTC provisions. In a reform file, I basically set all the ctc provisions from [ARPA.json](https://github.com/PSLmodels/examples/blob/main/psl_examples/taxcalc/ARPA.json) to 0 or false for 2021. 

`{
    "CTC_new_c": {"2021": 0.00},
    "CTC_new_c_under6_bonus": {"2021": 0.00},
    "CTC_include17": {"2021": false},
    "CTC_new_for_all": {"2021": false},
    "CTC_new_ps": {"2021": [0, 0, 0, 0, 0]},
    "CTC_new_prt": {"2021": 0},
    "CTC_refundable": {"2021": false}
}`

I also made the CTC reform explicit to make sure I didn't encounter issue the $1000 vs. $600 bonus issue, though I see that's been fixed (thanks!). So my CTC status-quo file looks like this:

`{
    "CTC_new_c": {"2021": 1000.00},
    "CTC_new_c_under6_bonus": {"2021": 600.00},
    "CTC_include17": {"2021": true},
    "CTC_new_for_all": {"2021": true},
    "CTC_new_ps": {"2021": [75000, 150000, 75000, 112500, 150000]},
    "CTC_new_prt": {"2021": 0.05},
    "CTC_refundable": {"2021": true}
}`

My terminal commands look like this:
` tc cps_2019_for_taxsim.csv 2021 --dump --reform ctc-repealed.json`
and 
` tc cps_2019_for_taxsim.csv 2021 --dump --reform ctc-new-current-law.json`

Comparing the change in aftertax income from the reforms, the results don't seem quite right, with sharp phase-outs at AGIs <$50,000. I have no idea what could be driving that behavior. Below are married filing jointly units with 1 child under 18 to give a sense of the issue. 

![image](https://user-images.githubusercontent.com/19849462/127692500-7504b686-4087-474a-9937-1b1de14f5242.png)

I tried setting all the ACTC parameters to zero and that didn't change anything. Is there anything obvious I'm doing wrong? I will try to make a minimal reproducible example of a datafile if it's helpful.

**Comment by JackLandry** [comment]
Here's a test file I ran that illustrates the issue
[test_ctc_file.csv](https://github.com/PSLmodels/Tax-Calculator/files/6910041/test_ctc_file.csv)

Here's the code creating the file

```
#Creating the test file
library(tidyverse)
library(here)
df <- tibble(e00200=seq(from = 0, to = 200000, by = 1000))
df <- df %>% mutate(RECID=row_number(), MARS=1, nu18=1, n24=1, nu06=0, FLPDYR=2021, age_head=50,
                    e00200p=e00200, XTOT=3)
write_csv(df, here("tax-calculator-code","test_ctc_file.csv"))

#Reading in results after running though tax calculator
new_ctc <- read_csv(here("tax-calculator-code","test_ctc_file-21-#-ctc-new-current-law-#.csv"))
no_new_ctc <- read_csv(here("tax-calculator-code","test_ctc_file-21-#-ctc-repealed-#.csv"))

#Getting results setup for graphing
new_ctc <- new_ctc %>% 
  rename(aftertax_income_ctc=aftertax_income)
no_new_ctc <- no_new_ctc %>% select(RECID,aftertax_income) %>% 
  rename(aftertax_income_no_ctc=aftertax_income)

tax_reform <- new_ctc %>% left_join(no_new_ctc)
tax_reform <- tax_reform %>% mutate(change_aftertax_income=aftertax_income_ctc-aftertax_income_no_ctc)

#Graph of result
ggplot(tax_reform,
       aes(x = c00100, y = change_aftertax_income)) +
  geom_point() +
  labs(y = "New\nCTC\nCredit", x = "Adjusted Gross Income") +
  scale_y_continuous(labels=scales::dollar_format()) +
  scale_x_continuous(labels=scales::dollar_format()) +
  theme_minimal()

```

And the resulting graph
![image](https://user-images.githubusercontent.com/19849462/127712438-4fe7081b-ba34-45db-8f66-1238fd060d8c.png)

**Comment by martinholmer** [comment]
@JackLandry, in your first comment in issue #2614 you say:

> Below are married filing jointly units with 1 child under 18 to give a sense of the issue.

and then you say:

> I will try to make a minimal reproducible example of a datafile if it's helpful.

Then in your second comment (which I assume is the "minimal reproducible example"), you set MARS=1.   But a married couple filing jointly is specified by MARS=2.  (MARS=1 designates a single individual.)

Is the graph you show at the end of your second comment not what you expect?  If it is not what you expect, how does the shown graph differ from what you expect?

By the way,  I like your use of R to generate the custom model input file and to analyze and graph the two model output files.

**Comment by martinholmer** [comment]
@JackLandry,  I've revised your `test_ctc_file.csv` by setting MARS=2 (instead of 1) for all the filing units.  This means Tax-Calculator interprets each case as a married couple with one child, which I think was your intention. 

 I used your two JSON reform files with no changes, and I executed the Tax-Calculator CLI `tc` twice calling for dump output in both runs, just as you did.  I then put the two
dump output CSV files into a single SQLite database with two tables: `noctc` and
`ctc21`.    And finally,  using a couple of simple SQL statements, I constructed the 
following table containing `ctc21` minus `noctc` changes for aftertax income and
four component variables:

<pre>
RECID  earnings  atinc_chg  ctc_new_chg  odc_chg  ctc_old_chg  actc_old_chg
-----  --------  ---------  -----------  -------  -----------  ------------
1      0.0       3000.0     1000.0       0.0      2000.0       0.0         
2      1000.0    3000.0     1000.0       0.0      2000.0       0.0         
3      2000.0    3000.0     1000.0       0.0      2000.0       0.0         
4      3000.0    2925.0     1000.0       0.0      2000.0       -75.0       
5      4000.0    2775.0     1000.0       0.0      2000.0       -225.0      
6      5000.0    2625.0     1000.0       0.0      2000.0       -375.0      
7      6000.0    2475.0     1000.0       0.0      2000.0       -525.0      
8      7000.0    2325.0     1000.0       0.0      2000.0       -675.0      
9      8000.0    2175.0     1000.0       0.0      2000.0       -825.0      
10     9000.0    2025.0     1000.0       0.0      2000.0       -975.0      
11     10000.0   1875.0     1000.0       0.0      2000.0       -1125.0     
12     11000.0   1725.0     1000.0       0.0      2000.0       -1275.0     
13     12000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
14     13000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
15     14000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
16     15000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
17     16000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
18     17000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
19     18000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
20     19000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
21     20000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
22     21000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
23     22000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
24     23000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
25     24000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
26     25000.0   1600.0     1000.0       0.0      2000.0       -1400.0     
27     26000.0   1502.0     1000.0       0.0      1902.0       -1400.0     
28     27000.0   1402.0     1000.0       0.0      1802.0       -1400.0     
29     28000.0   1302.0     1000.0       0.0      1702.0       -1400.0     
30     29000.0   1202.0     1000.0       0.0      1602.0       -1400.0     
31     30000.0   1102.0     1000.0       0.0      1502.0       -1400.0     
32     31000.0   1002.0     1000.0       0.0      1402.0       -1400.0     
33     32000.0   1000.0     1000.0       0.0      1302.0       -1302.0     
34     33000.0   1000.0     1000.0       0.0      1202.0       -1202.0     
35     34000.0   1000.0     1000.0       0.0      1102.0       -1102.0     
36     35000.0   1000.0     1000.0       0.0      1002.0       -1002.0     
37     36000.0   1000.0     1000.0       0.0      902.0        -902.0      
38     37000.0   1000.0     1000.0       0.0      802.0        -802.0      
39     38000.0   1000.0     1000.0       0.0      702.0        -702.0      
40     39000.0   1000.0     1000.0       0.0      602.0        -602.0      
41     40000.0   1000.0     1000.0       0.0      502.0        -502.0      
42     41000.0   1000.0     1000.0       0.0      402.0        -402.0      
43     42000.0   1000.0     1000.0       0.0      302.0        -302.0      
44     43000.0   1000.0     1000.0       0.0      202.0        -202.0      
45     44000.0   1000.0     1000.0       0.0      102.0        -102.0      
46     45000.0   1000.0     1000.0       0.0      1.0          -1.0        
47     46000.0   1000.0     1000.0       0.0      0.0          0.0         
48     47000.0   1000.0     1000.0       0.0      0.0          0.0         
49     48000.0   1000.0     1000.0       0.0      0.0          0.0         
50     49000.0   1000.0     1000.0       0.0      0.0          0.0         
51     50000.0   1000.0     1000.0       0.0      0.0          0.0         
52     51000.0   1000.0     1000.0       0.0      0.0          0.0         
53     52000.0   1000.0     1000.0       0.0      0.0          0.0         
54     53000.0   1000.0     1000.0       0.0      0.0          0.0         
55     54000.0   1000.0     1000.0       0.0      0.0          0.0         
56     55000.0   1000.0     1000.0       0.0      0.0          0.0         
57     56000.0   1000.0     1000.0       0.0      0.0          0.0         
58     57000.0   1000.0     1000.0       0.0      0.0          0.0         
59     58000.0   1000.0     1000.0       0.0      0.0          0.0         
60     59000.0   1000.0     1000.0       0.0      0.0          0.0         
61     60000.0   1000.0     1000.0       0.0      0.0          0.0         
62     61000.0   1000.0     1000.0       0.0      0.0          0.0         
63     62000.0   1000.0     1000.0       0.0      0.0          0.0         
64     63000.0   1000.0     1000.0       0.0      0.0          0.0         
65     64000.0   1000.0     1000.0       0.0      0.0          0.0         
66     65000.0   1000.0     1000.0       0.0      0.0          0.0         
67     66000.0   1000.0     1000.0       0.0      0.0          0.0         
68     67000.0   1000.0     1000.0       0.0      0.0          0.0         
69     68000.0   1000.0     1000.0       0.0      0.0          0.0         
70     69000.0   1000.0     1000.0       0.0      0.0          0.0         
71     70000.0   1000.0     1000.0       0.0      0.0          0.0         
72     71000.0   1000.0     1000.0       0.0      0.0          0.0         
73     72000.0   1000.0     1000.0       0.0      0.0          0.0         
74     73000.0   1000.0     1000.0       0.0      0.0          0.0         
75     74000.0   1000.0     1000.0       0.0      0.0          0.0         
76     75000.0   1000.0     1000.0       0.0      0.0          0.0         
77     76000.0   1000.0     1000.0       0.0      0.0          0.0         
78     77000.0   1000.0     1000.0       0.0      0.0          0.0         
79     78000.0   1000.0     1000.0       0.0      0.0          0.0         
80     79000.0   1000.0     1000.0       0.0      0.0          0.0         
81     80000.0   1000.0     1000.0       0.0      0.0          0.0         
82     81000.0   1000.0     1000.0       0.0      0.0          0.0         
83     82000.0   1000.0     1000.0       0.0      0.0          0.0         
84     83000.0   1000.0     1000.0       0.0      0.0          0.0         
85     84000.0   1000.0     1000.0       0.0      0.0          0.0         
86     85000.0   1000.0     1000.0       0.0      0.0          0.0         
87     86000.0   1000.0     1000.0       0.0      0.0          0.0         
88     87000.0   1000.0     1000.0       0.0      0.0          0.0         
89     88000.0   1000.0     1000.0       0.0      0.0          0.0         
90     89000.0   1000.0     1000.0       0.0      0.0          0.0         
91     90000.0   1000.0     1000.0       0.0      0.0          0.0         
92     91000.0   1000.0     1000.0       0.0      0.0          0.0         
93     92000.0   1000.0     1000.0       0.0      0.0          0.0         
94     93000.0   1000.0     1000.0       0.0      0.0          0.0         
95     94000.0   1000.0     1000.0       0.0      0.0          0.0         
96     95000.0   1000.0     1000.0       0.0      0.0          0.0         
97     96000.0   1000.0     1000.0       0.0      0.0          0.0         
98     97000.0   1000.0     1000.0       0.0      0.0          0.0         
99     98000.0   1000.0     1000.0       0.0      0.0          0.0         
100    99000.0   1000.0     1000.0       0.0      0.0          0.0         
101    100000.0  1000.0     1000.0       0.0      0.0          0.0         
102    101000.0  1000.0     1000.0       0.0      0.0          0.0         
103    102000.0  1000.0     1000.0       0.0      0.0          0.0         
104    103000.0  1000.0     1000.0       0.0      0.0          0.0         
105    104000.0  1000.0     1000.0       0.0      0.0          0.0         
106    105000.0  1000.0     1000.0       0.0      0.0          0.0         
107    106000.0  1000.0     1000.0       0.0      0.0          0.0         
108    107000.0  1000.0     1000.0       0.0      0.0          0.0         
109    108000.0  1000.0     1000.0       0.0      0.0          0.0         
110    109000.0  1000.0     1000.0       0.0      0.0          0.0         
111    110000.0  1000.0     1000.0       0.0      0.0          0.0         
112    111000.0  1000.0     1000.0       0.0      0.0          0.0         
113    112000.0  1000.0     1000.0       0.0      0.0          0.0         
114    113000.0  1000.0     1000.0       0.0      0.0          0.0         
115    114000.0  1000.0     1000.0       0.0      0.0          0.0         
116    115000.0  1000.0     1000.0       0.0      0.0          0.0         
117    116000.0  1000.0     1000.0       0.0      0.0          0.0         
118    117000.0  1000.0     1000.0       0.0      0.0          0.0         
119    118000.0  1000.0     1000.0       0.0      0.0          0.0         
120    119000.0  1000.0     1000.0       0.0      0.0          0.0         
121    120000.0  1000.0     1000.0       0.0      0.0          0.0         
122    121000.0  1000.0     1000.0       0.0      0.0          0.0         
123    122000.0  1000.0     1000.0       0.0      0.0          0.0         
124    123000.0  1000.0     1000.0       0.0      0.0          0.0         
125    124000.0  1000.0     1000.0       0.0      0.0          0.0         
126    125000.0  1000.0     1000.0       0.0      0.0          0.0         
127    126000.0  1000.0     1000.0       0.0      0.0          0.0         
128    127000.0  1000.0     1000.0       0.0      0.0          0.0         
129    128000.0  1000.0     1000.0       0.0      0.0          0.0         
130    129000.0  1000.0     1000.0       0.0      0.0          0.0         
131    130000.0  1000.0     1000.0       0.0      0.0          0.0         
132    131000.0  1000.0     1000.0       0.0      0.0          0.0         
133    132000.0  1000.0     1000.0       0.0      0.0          0.0         
134    133000.0  1000.0     1000.0       0.0      0.0          0.0         
135    134000.0  1000.0     1000.0       0.0      0.0          0.0         
136    135000.0  1000.0     1000.0       0.0      0.0          0.0         
137    136000.0  1000.0     1000.0       0.0      0.0          0.0         
138    137000.0  1000.0     1000.0       0.0      0.0          0.0         
139    138000.0  1000.0     1000.0       0.0      0.0          0.0         
140    139000.0  1000.0     1000.0       0.0      0.0          0.0         
141    140000.0  1000.0     1000.0       0.0      0.0          0.0         
142    141000.0  1000.0     1000.0       0.0      0.0          0.0         
143    142000.0  1000.0     1000.0       0.0      0.0          0.0         
144    143000.0  1000.0     1000.0       0.0      0.0          0.0         
145    144000.0  1000.0     1000.0       0.0      0.0          0.0         
146    145000.0  1000.0     1000.0       0.0      0.0          0.0         
147    146000.0  1000.0     1000.0       0.0      0.0          0.0         
148    147000.0  1000.0     1000.0       0.0      0.0          0.0         
149    148000.0  1000.0     1000.0       0.0      0.0          0.0         
150    149000.0  1000.0     1000.0       0.0      0.0          0.0         
151    150000.0  1000.0     1000.0       0.0      0.0          0.0         
152    151000.0  950.0      950.0        0.0      0.0          0.0         
153    152000.0  900.0      900.0        0.0      0.0          0.0         
154    153000.0  850.0      850.0        0.0      0.0          0.0         
155    154000.0  800.0      800.0        0.0      0.0          0.0         
156    155000.0  750.0      750.0        0.0      0.0          0.0         
157    156000.0  700.0      700.0        0.0      0.0          0.0         
158    157000.0  650.0      650.0        0.0      0.0          0.0         
159    158000.0  600.0      600.0        0.0      0.0          0.0         
160    159000.0  550.0      550.0        0.0      0.0          0.0         
161    160000.0  500.0      500.0        0.0      0.0          0.0         
162    161000.0  450.0      450.0        0.0      0.0          0.0         
163    162000.0  400.0      400.0        0.0      0.0          0.0         
164    163000.0  350.0      350.0        0.0      0.0          0.0         
165    164000.0  300.0      300.0        0.0      0.0          0.0         
166    165000.0  250.0      250.0        0.0      0.0          0.0         
167    166000.0  200.0      200.0        0.0      0.0          0.0         
168    167000.0  150.0      150.0        0.0      0.0          0.0         
169    168000.0  100.0      100.0        0.0      0.0          0.0         
170    169000.0  50.0       50.0         0.0      0.0          0.0         
171    170000.0  0.0        0.0          0.0      0.0          0.0         
172    171000.0  0.0        0.0          0.0      0.0          0.0         
173    172000.0  0.0        0.0          0.0      0.0          0.0         
174    173000.0  0.0        0.0          0.0      0.0          0.0         
175    174000.0  0.0        0.0          0.0      0.0          0.0         
176    175000.0  0.0        0.0          0.0      0.0          0.0         
177    176000.0  0.0        0.0          0.0      0.0          0.0         
178    177000.0  0.0        0.0          0.0      0.0          0.0         
179    178000.0  0.0        0.0          0.0      0.0          0.0         
180    179000.0  0.0        0.0          0.0      0.0          0.0         
181    180000.0  0.0        0.0          0.0      0.0          0.0         
182    181000.0  0.0        0.0          0.0      0.0          0.0         
183    182000.0  0.0        0.0          0.0      0.0          0.0         
184    183000.0  0.0        0.0          0.0      0.0          0.0         
185    184000.0  0.0        0.0          0.0      0.0          0.0         
186    185000.0  0.0        0.0          0.0      0.0          0.0         
187    186000.0  0.0        0.0          0.0      0.0          0.0         
188    187000.0  0.0        0.0          0.0      0.0          0.0         
189    188000.0  0.0        0.0          0.0      0.0          0.0         
190    189000.0  0.0        0.0          0.0      0.0          0.0         
191    190000.0  0.0        0.0          0.0      0.0          0.0         
192    191000.0  0.0        0.0          0.0      0.0          0.0         
193    192000.0  0.0        0.0          0.0      0.0          0.0         
194    193000.0  0.0        0.0          0.0      0.0          0.0         
195    194000.0  0.0        0.0          0.0      0.0          0.0         
196    195000.0  0.0        0.0          0.0      0.0          0.0         
197    196000.0  0.0        0.0          0.0      0.0          0.0         
198    197000.0  0.0        0.0          0.0      0.0          0.0         
199    198000.0  0.0        0.0          0.0      0.0          0.0         
200    199000.0  0.0        0.0          0.0      0.0          0.0         
201    200000.0  0.0        0.0          0.0      0.0          0.0         
</pre>

Given my limited knowledge of the new CTC provisions for 2021, it seems as if the phase-out of the CTC_new amount from $1000 down to zero is sensible.  You are obviously more knowledgeable about the new policy (as you recently found a related Tax-Calculator bug).  Does the CTC_new phase-out meet your expectations?  Or is it "weird"?

The puzzling changes in aftertax income that you identified in your earlier comments in this issue are caused by changes in the old (non-refundable) CTC amount and in the 
old (refundable) ACTC amount.  Are the changes in those amounts (show in the table
above) what you would expect?

If you are interested in the details of what I've done, I'd be happy to share them.
